File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> nodes;
        dfs(root, nodes);
        return nodes;
    }

    void dfs(TreeNode* node, vector<int>& nodes) {
        if (node == nullptr) return;
        dfs(node->left, nodes);
        dfs(node->right, nodes);
        nodes.push_back(node->val);
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(node, maxi):
            if not node: return 0
            left = dfs(node.left, max(maxi, node.val))
            right = dfs(node.right, max(maxi, node.val))
            return left+right if node.val < maxi else left+right+1
        return dfs(root, -101)